### PR TITLE
button-storybook

### PR DESCRIPTION
**Pull request by MarcelloAragoni**
**Trello:** https://trello.com/c/Iav0wXku/8-todo-list

## Description

Created button component, and implemented storybook.

## Screenshot
image

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const parameters = {
```
Comment (ralfting) [inline .storybook/preview.js]: I that this file could be removed for now.

File: package.json [hunk 2]
```diff
@@ -14,19 +14,32 @@
...
     "web-vitals": "^2.1.4"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "storybook": "start-storybook -p 6006 -s public",
+    "build-storybook": "build-storybook -s public"
```
Comment (ralfting) [inline package.json:27]: Well done!

File: package.json [hunk 3]
```diff
@@ -40,5 +53,20 @@
...
+    "@storybook/addon-links": "^6.5.9",
+    "@storybook/builder-webpack5": "^6.5.9",
+    "@storybook/manager-webpack5": "^6.5.9",
+    "@storybook/node-logger": "^6.5.9",
+    "@storybook/preset-create-react-app": "^4.1.2",
+    "@storybook/react": "^6.5.9",
+    "@storybook/testing-library": "^0.0.13",
+    "babel-plugin-named-exports-order": "^0.0.2",
+    "prop-types": "^15.8.1",
+    "webpack": "^5.73.0"
```
Comment (ralfting) [inline package.json]: I think that you can get rid of this one, React Create App already has a `webpack` inside of it.

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import cn from "classnames";
+
+import "./button.scss";
+
+export default function Button({ label, color, onClick, variant }:any) {
```
Comment (ralfting) [inline src/components/Button/Button.tsx]: cool, now let's add some type in here:

1. In order to define a type for the props received (Props) you will replace `any` for the type that you want to, by default I use `Props` (type name is always in CapitalizeCase)

2. To define the type you use the keyword `type` instead `const`, like so:
```ts
type Props = {

}
```

3. You'll see where you're using the props and how to determinate which type is:
`color` is a **string**, `variant` is a **string**, `label` is a **string**, and`onClick` is a **function**.

So our type will look like:
```ts
type Props = {
  color: string;
  label: string;
  variant: string;
  onClick: () => void
}
```

4. Finally you'll be able to attach is to your component props.
```suggestion
export default function Button({ label, color, onClick, variant }: Props) {
```

File: src/components/Button/button.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  padding: 0 20px;
+  font-weight: bold;
+  background-repeat: no-repeat;
+  background-position: center;
+  color: black;
+  background-color: transparent;
+  cursor:pointer;
+
+  &--ghost{
+    padding: 0;
```
Comment (ralfting) [inline src/components/Button/button.scss]: is it possible to use the same padding as the `.button`?

File: src/components/Button/button.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    padding-left: 30px;
+    padding-right: 35px;
+    border: none;
+  }
+
+  &--blue{
+    color: hsl(234, 11%, 52%);
```
Comment (ralfting) [inline src/components/Button/button.scss]: I recommend to create a scss variable to manage the colors, is it hard to know which color is this one 🤔 

Try to set on top of this file a variable like so:
```scss
$gray: hsl(234, 11%, 52%); // I have no idea what color is this haha
```

and then use it in the color property:
```suggestion
    color: $gray;
```

File: package.json (hunk 3 above)
Comment (MarcelloAragoni) [reply package.json]: about this one, do i remove this 4 lines?

Comment (ralfting) [reply package.json]: only the `webpack`, use the npm to perform it:

```bash
npm uninstall webpack
```

if you do not use `npm` to do that, may you'll face some problems of version in the package.

**Review by ralfting — APPROVED** [review]

**Pull request merged**